### PR TITLE
fix: iwyu header fixes

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
Automatic fixes obtained with:
```
iwyu_tool -p build -o iwyu -- -Xiwyu --cxx17ns -Xiwyu --mapping_file=$PWD/.github/iwyu.imp | fix_include --blank_lines --nosafe_headers --reorder --separate_project_includes="<tld>" --keep_iwyu_namespace_format
```

### What kind of change does this PR introduce?
- [x] Bug fix (issue: header cleanup)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

### CODE REVIEW

**Comment by veprbl** [comment]
Regarding introduction of the forward declarations. The https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/WhyIWYU.md#why-forward-declare suggests that in additional to parsing speed win, it may allow to avoid some ODR rule violations, which I don't think is as robust. As this is a developing codebase, it would be easier for users to having to pick and switch between forward declaration and full include depending on whether they use any methods/members from a class or not.

**Comment by wdconinc** [comment]
The problem I see with fwd decl is mainly one of developer education: when do you need to include the full header, when do you include the fwd header (spdlog/fwd.h), and when do you just use a fwd decl yourself? I think that because this tool won't give suggested changes like clang-tidy does it will appear as magic to some, which will lead to frustration. Not a reason not to do it, but we need to think about how to train people and possibly how to make the iwyu suggestions clear. Right now I don't have a good solution for that (short of using capybara to create a PR against the PR).

File: src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc [hunk 1]
```diff
@@ -1,22 +1,48 @@
 // SPDX-License-Identifier: LGPL-3.0-or-later
 // Copyright (C) 2022 Sylvester Joosten, Chao, Chao Peng, Whitney Armstrong, Dhevan Gangadharan
 
+#include <DD4hep/Detector.h>
+#include <Eigen/src/Eigenvalues/EigenSolver.h>
+#include <Eigen/src/Householder/BlockHouseholder.h>
+#include <Eigen/src/Householder/Householder.h>
+#include <Eigen/src/Jacobi/Jacobi.h>
```
Comment (wdconinc) [inline src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc:8]: Not sure why this isn't caught by
```
{ include: ["@<Eigen/src/Householder/.*>", "private", "<Eigen/Householder>", "public"] },
```

**Comment by veprbl** [comment]
Yes. This is basically my point, we'd confuse people unnecessarily. We can always enable forward declarations later.

File: src/algorithms/calorimetry/CalorimeterHitDigi.h [hunk 2]
```diff
@@ -13,17 +13,17 @@
 
 #pragma once
 
+#include <stdint.h>
```
Comment (wdconinc) [inline src/algorithms/calorimetry/CalorimeterHitDigi.h:16]: Boost dislikes `cstdint`

File: src/algorithms/calorimetry/CalorimeterHitsMerger.cc [hunk 3]
```diff
@@ -10,6 +10,32 @@
...
+#include <DD4hep/Detector.h>
+#include <DD4hep/IDDescriptor.h>
+#include <DD4hep/Objects.h>
+#include <DD4hep/Readout.h>
+#include <DD4hep/VolumeManager.h>
+#include <DDRec/CellIDPositionConverter.h>
+#include <DDSegmentation/BitFieldCoder.h>
+#include <Evaluator/DD4hepUnits.h>
+#include <Math/GenVector/DisplacementVector3D.h>
+#include <bits/utility.h>
```
Comment (wdconinc) [inline src/algorithms/calorimetry/CalorimeterHitsMerger.cc:24]: Avoid low level headers

File: src/algorithms/tracking/IterativeVertexFinder.cc [hunk 4]
```diff
@@ -5,32 +5,51 @@
...
 #include <Acts/Vertexing/HelicalTrackLinearizer.hpp>
+#include <Acts/Vertexing/HelicalTrackLinearizer.ipp>
 #include <Acts/Vertexing/ImpactPointEstimator.hpp>
+#include <Acts/Vertexing/ImpactPointEstimator.ipp>
 #include <Acts/Vertexing/IterativeVertexFinder.hpp>
-#include <Acts/Vertexing/LinearizedTrack.hpp>
+#include <Acts/Vertexing/IterativeVertexFinder.ipp>
 #include <Acts/Vertexing/Vertex.hpp>
-#include <Acts/Vertexing/VertexFinderConcept.hpp>
+#include <Acts/Vertexing/Vertex.ipp>
```
Comment (wdconinc) [inline src/algorithms/tracking/IterativeVertexFinder.cc:29]: Avoid ipp headers.

**Comment by wdconinc** [comment]
Closing in favor of new PR.

**Pull request closed without merging**